orbitdeterminator/doppler/utils/utils_aux.py

```diff
@@ -0,0 +1,406 @@
+    v_teme = CartesianDifferential(v[:,0], v[:,1], v[:,2], unit=u.km/u.s)
+
+    # Temporary workaround until astropy version 4.1 that supports TEME
+    astropy_version = float(astropy.__version__[0:3])
+
+    if astropy_version < 4.1:
+        print(f"Warning: astropy version {astropy_version} < 4.1, treating SGP4 output (TEME) as FK5")
+        eci = FK5(r_teme.with_differentials(v_teme), obstime=t)
+        frame = 'fk5'
+    else:
+        eci = TEME(r_teme.with_differentials(v_teme), obstime=t)
+
+    # Coordinate frame transformations
+    if frame=='teme':
+        x_sat = np.array([eci.x.value, eci.y.value, eci.z.value, 
+                                eci.v_x.value, eci.v_y.value, eci.v_z.value])
+
+    if frame=='fk5':
+        # If the astropy version < 4.1, keep it there
+        if astropy_version < 4.1:
+            x_sat = np.array([eci.x.value, eci.y.value, eci.z.value, 
+                                eci.v_x.value, eci.v_y.value, eci.v_z.value])
+        # If the astropy vesion >= 4.1, transform TEME to FK5
+        else:
+            fk5 = eci.transform_to(FK5(obstime=t))
+            x_sat = np.array([fk5.x.value, fk5.y.value, fk5.z.value, 
+                                fk5.v_x.value, fk5.v_y.value, fk5.v_z.value])
+
+    elif frame=='itrs':
+        itrs = eci.transform_to(ITRS(obstime=t))
+        x_sat = np.array([itrs.x.value, itrs.y.value, itrs.z.value, 
+                                itrs.v_x.value, itrs.v_y.value, itrs.v_z.value])
+
+    return x_sat, t
+
+def get_site(lat, lon, height, obstime, frame='teme'):
+    """ Auxiliary function to obtain site coordinates in ITRS or TEME frame.
+
+    Args:
+        lat (float): latitude (degrees).
+        lon (float): longitude (degrees).
+        height (float): altitude (m).
+        obstime (astropy.time.Time): time array (n, ).
+        frame (str): frame (teme or itrs). Defaults to 'teme'.
+    Returns:
+        x_obs (np.ndarray): array with site positions in ITRS/TEME frame (6, n).
+    """
+
+    v = np.zeros(obstime.shape[0])      # Temporary variable
+
+    # Switch to FK5 if astropy version doesn't support TEME frame
+    if float(astropy.__version__[0:3]) < 4.1:
+        frame='fk5'
+
+
+    if frame == 'itrs':
+        site = EarthLocation(lat=lat*u.deg, lon=lon*u.deg, height=height*u.m)
+        site_itrs_temp = site.get_itrs(obstime=obstime)
+
+        x_obs = np.array([site_itrs_temp.x.value, site_itrs_temp.y.value, site_itrs_temp.z.value,
+            v, v, v])
+            
+    elif frame == 'teme':
+        # Need some workaround conversions for TEME frame
+        site = EarthLocation(lat=lat*u.deg, lon=lon*u.deg, height=height/1e3*u.km)
+        site_itrs_temp = site.get_itrs(obstime=obstime)
+
+        r_itrs = CartesianRepresentation(
+            site_itrs_temp.data.xyz.value[0,:], 
+            site_itrs_temp.data.xyz.value[1,:], 
+            site_itrs_temp.data.xyz.value[2,:], unit=u.km)
```

Comment:
I'd say that

```suggestion
        r_itrs = site_itrs_temp.cartesian
```

does the same (see other instances of this below).